### ISSUE TITLE
error loading binary mask TIFF images

### ISSUE THREAD

**Issue by wyli** [NifTK/NiftyNet#268]
1. What you were trying to do (and why)
loading binary mask `.tif` image with `loader=pillow`

2. What happened (include command output)
```
Number of subjects 2, input section names: ['subject_id', 'mr', 'label']
-- using all subjects (without data partitioning).

Traceback (most recent call last):--| 50.0% 
  File "test.py", line 27, in <module>
    "loader": 'pillow'
  File "/usr/local/anaconda3/lib/python3.6/site-packages/niftynet/io/image_reader.py", line 173, in initialise
    file_list, self._input_sources, data_param)
  File "/usr/local/anaconda3/lib/python3.6/site-packages/niftynet/io/image_reader.py", line 403, in _filename_to_image_list
    file_list, idx, modalities, data_param)
  File "/usr/local/anaconda3/lib/python3.6/site-packages/niftynet/io/image_reader.py", line 470, in _create_image
    return ImageFactory.create_instance(**image_properties)
  File "/usr/local/anaconda3/lib/python3.6/site-packages/niftynet/io/image_type.py", line 658, in create_instance
    ndims = misc.infer_ndims_from_file(file_path[0], loader[0])
  File "/usr/local/anaconda3/lib/python3.6/site-packages/niftynet/io/misc_io.py", line 44, in infer_ndims_from_file
    image_header = load_image_obj(file_path, loader).header
  File "/usr/local/anaconda3/lib/python3.6/site-packages/niftynet/io/image_loader.py", line 96, in load_image_obj
    return loader_params['func'](filename)
  File "/usr/local/anaconda3/lib/python3.6/site-packages/niftynet/io/image_loader.py", line 156, in imread_pillow
    return image2nibabel(img)
  File "/usr/local/anaconda3/lib/python3.6/site-packages/niftynet/io/image_loader.py", line 194, in image2nibabel
    return ImageAsNibabel(img, affine)
  File "/usr/local/anaconda3/lib/python3.6/site-packages/niftynet/io/image_loader.py", line 210, in __init__
    nib.Nifti1Image.__init__(self, img, affine)
  File "/usr/local/anaconda3/lib/python3.6/site-packages/nibabel/nifti1.py", line 1762, in __init__
    file_map)
  File "/usr/local/anaconda3/lib/python3.6/site-packages/nibabel/analyze.py", line 923, in __init__
    dataobj, affine, header, extra, file_map)
  File "/usr/local/anaconda3/lib/python3.6/site-packages/nibabel/spatialimages.py", line 371, in __init__
    self._header.set_data_dtype(dataobj.dtype)
  File "/usr/local/anaconda3/lib/python3.6/site-packages/nibabel/analyze.py", line 578, in set_data_dtype
    'data dtype "{0}" not supported'.format(datatype))
nibabel.spatialimages.HeaderDataError: data dtype "bool" not supported
```
3. What you expected to happen
loading the mask into array of integers

**Comment by ericspod** [comment]
@wyli: this doesn't appear to have been fixed in the branches referencing #268.

**Comment by wyli** [comment]
> @wyli: this doesn't appear to have been fixed in the branches referencing #268.

@ericspod No, this hasn't been fixed.
The commit message is causing the confusion here --
messages created before June 2018 may contain issue/PR id such as `#268`, 
refer to our previous cmiclab ticket (`https://cmiclab.cs.ucl.ac.uk/CMIC/NiftyNet/issues/268`). here the ids are automatically linked to github (`https://github.com/NifTK/NiftyNet/issues/268`), which is incorrect because the issue lists are never synced.
